core/store/src/main/java/org/locationtech/geowave/core/store/base/BaseDataStore.java

```diff
@@ -1166,42 +1199,347 @@ private void internalAddIndices(
+						// others
+						markedAdapters.add(adapterIndexMap.getAdapterId());
+						continue;
+					}
+				}
+			}
+		}
+		it.close();
+
+		// take out the index from the data statistics, and mapping
+		for (int i = 0; i < markedAdapters.size(); i++) {
+			final short adapterId = markedAdapters.get(i);
+			baseOperations.deleteAll(
+					indexName,
+					"",
+					adapterId);
+			statisticsStore.removeAllStatistics(adapterId);
+			indexMappingStore.remove(
+					adapterId,
+					indexName);
+		}
+		// remove the actual index
+		indexStore.removeIndex(indexName);
 	}
 
 	@Override
 	public void removeIndex(
 			final String typeName,
-			final String indexName ) {
-		// TODO Auto-generated method stub
+			final String indexName )
+			throws IllegalStateException {
+
+		// First make sure the adapter exists and that this is not the only
+		// index left for the given adapter. If it is, we should throw an
+		// exception.
+		final short adapterId = internalAdapterStore.getAdapterId(typeName);
+		final AdapterToIndexMapping adapterIndexMap = indexMappingStore.getIndicesForAdapter(adapterId);
+
+		if (adapterIndexMap == null) {
+			throw new IllegalArgumentException(
+					"No adapter with typeName " + typeName + "could be found.");
+		}
+
+		final String[] indexNames = adapterIndexMap.getIndexNames();
+		if (indexNames.length == 1) {
+			throw new IllegalStateException(
+					"Index removal failed. Adapters require at least one index.");
+		}
 
+		// Remove all the data for the adapter and index
+		baseOperations.deleteAll(
+				indexName,
+				"",
+				adapterId);
+
+		// If this is the last adapter/type associated with the index, then we
+		// can remove the actual index too.
+		Short[] adapters = getAdaptersForIndex(indexName);
+		if (adapters.length == 1) indexStore.removeIndex(indexName);
+
+		// Finally, remove the mapping
+		indexMappingStore.remove(
+				adapterId,
+				indexName);
 	}
 
 	@Override
 	public void removeType(
 			final String typeName ) {
-		// TODO Auto-generated method stub
+		// Removing a type requires removing the data associated with the type,
+		// the index mapping for the type, and we also need to remove stats for
+		// the type.
+		final Short adapterId = internalAdapterStore.getAdapterId(typeName);
 
+		if (adapterId != -1) {
+			final AdapterToIndexMapping mapping = indexMappingStore.getIndicesForAdapter(adapterId);
+			final String[] indexNames = mapping.getIndexNames();
+
+			// remove all the data for each index paired to this adapter
+			for (int i = 0; i < indexNames.length; i++) {
+				baseOperations.deleteAll(
+						indexNames[i],
+						"",
+						adapterId);
+			}
+
+			statisticsStore.removeAllStatistics(adapterId);
+			indexMappingStore.remove(adapterId);
+			internalAdapterStore.remove(adapterId);
+			adapterStore.removeAdapter(adapterId);
+		}
 	}
 
 	@Override
 	public void deleteAll() {
-		deleteEverything();
+		// Clear all the data and stats, but leave adapters/indices alone
```

Comment:
is that the desired behavior?  it is nice to have a "wipe everything" op as well then